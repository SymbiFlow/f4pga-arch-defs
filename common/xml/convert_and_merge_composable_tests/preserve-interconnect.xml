<?xml version="1.0"?>
<!-- Make sure existing interconnect values are preserved. -->
<xml>
  <pb_type name="parent">
    <input name="ia1"/>
    <input name="ia2"/>
    <input name="ia3"/>
    <output name="o0"/>
    <output name="o1"/>
    <pb_type name="childa">
      <input name="i1"/>
      <input name="i2"/>
      <output name="o"/>
    </pb_type>
    <pb_type name="childb">
      <input name="i"/>
      <output name="o"/>
    </pb_type>
    <pb_type name="childc">
      <input name="i"/>
      <output name="o"/>
    </pb_type>
    <interconnect>
      <direct input="parent.ia1" name="childa-i1" output="childa.i1"/>
      <mux name="childa-input-i2" input="parent.ia2 parent.ia3" output="childa.i2">
	<pack_pattern name="MUX1" in_port="parent.ia2" output="childa.i2" />
	<pack_pattern name="MUX2" in_port="parent.ia3" output="childa.i2" />
      </mux>
      <direct input="childa.o" name="parent-o0" output="parent.o0">
	<pack_pattern name="CARRY" in_port="childa.o" output="parent.o0" />
      </direct>
      <direct input="childa.o" name="childb-i" output="childb.i"/>
      <mux name="childc-input" input="childa.o childb.o" output="childc.i"/>
      <mux name="output" input="childa.o childb.o childc.o" output="parent.o1"/>
    </interconnect>
  </pb_type>
</xml>
