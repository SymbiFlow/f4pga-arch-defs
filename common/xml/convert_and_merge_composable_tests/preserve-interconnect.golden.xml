<?xml version="1.0"?>
<xml>
  <pb_type name="parent">
    <input name="ia1"/>
    <input name="ia2"/>
    <input name="ia3"/>
    <output name="o0"/>
    <output name="o1"/>
    <pb_type name="childa">
      <input name="i1"/>
      <input name="i2"/>
      <output name="o"/>
    </pb_type>
    <pb_type name="childb">
      <input name="i"/>
      <output name="o"/>
    </pb_type>
    <pb_type name="childc">
      <input name="i"/>
      <output name="o"/>
    </pb_type>
    <interconnect>
      <direct input="parent.ia1" name="childa-i1" output="childa.i1"/>
      <mux input="parent.ia2 parent.ia3" name="childa-input-i2" output="childa.i2">
        <pack_pattern in_port="parent.ia2" name="MUX1" output="childa.i2"/>
        <pack_pattern in_port="parent.ia3" name="MUX2" output="childa.i2"/>
      </mux>
      <direct input="childa.o" name="childb-i" output="childb.i"/>
      <mux input="childa.o childb.o" name="childc-input" output="childc.i"/>
      <direct input="childa.o" name="parent-o0" output="parent.o0">
        <pack_pattern in_port="childa.o" name="CARRY" out_port="parent.o0" output="parent.o0"/>
      </direct>
      <mux input="childa.o childb.o childc.o" name="output" output="parent.o1"/>
    </interconnect>
  </pb_type>
</xml>
